test/secret/secret_m4_test.go

```diff
@@ -0,0 +1,142 @@
+		deployment, err = testenvInstance.NewDeployment(testenv.RandomDNSName(3))
+		Expect(err).To(Succeed(), "Unable to create deployment")
+	})
+
+	AfterEach(func() {
+		// When a test spec failed, skip the teardown so we can troubleshoot.
+		if CurrentGinkgoTestDescription().Failed {
+			testenvInstance.SkipTeardown = true
+		}
+		if deployment != nil {
+			deployment.Teardown()
+		}
+	})
+
+	Context("Multisite cluster deployment (M13 - Multisite indexer cluster, Search head cluster)", func() {
+		It("secret,integration: secret update on multisite indexers and search head cluster", func() {
+
+			/* Test Scenario
+			1. Update Secrets Data
+			2. Verify New versioned secret are created with correct value.
+			3. Verify new secrets are mounted on pods.
+			4. Verify New Secrets are present in server.conf (Pass4SymmKey) */
+
+			// Download License File
+			licenseFilePath, err := testenv.DownloadFromS3Bucket()
+			Expect(err).To(Succeed(), "Unable to download license file")
+
+			// Create License Config Map
+			testenvInstance.CreateLicenseConfigMap(licenseFilePath)
+
+			siteCount := 3
+			err = deployment.DeployMultisiteClusterWithSearchHead(deployment.GetName(), 1, siteCount)
+			Expect(err).To(Succeed(), "Unable to deploy cluster")
+
+			// Wait for License Master to be in READY status
+			testenv.LicenseMasterReady(deployment, testenvInstance)
+
+			// Ensure that the cluster-master goes to Ready phase
+			testenv.ClusterMasterReady(deployment, testenvInstance)
+
+			// Ensure the indexers of all sites go to Ready phase
+			testenv.IndexersReady(deployment, testenvInstance, siteCount)
+
+			// Ensure search head cluster go to Ready phase
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Ensure cluster configured as multisite
+			testenv.IndexerClusterMultisiteStatus(deployment, testenvInstance, siteCount)
+
+			// Verify MC Pod is Ready
+			testenv.MCPodReady(testenvInstance.GetName(), deployment)
+
+			// Get Current Secrets Struct
+			namespaceScopedSecretName := splcommon.GetNamespaceScopedSecretName(testenvInstance.GetName())
+			secretStruct, err := testenv.GetSecretStruct(deployment, testenvInstance.GetName(), namespaceScopedSecretName)
+			Expect(err).To(Succeed(), "Unable to get secret struct")
+
+			// Test 1
+			// Update Secrets Data and
+			// Verify New versioned secret are created with correct value.
+			// Verify new secrets are mounted on pods.
+			// Verify New Secrets are present in server.conf (Pass4SymmKey)
+
+			// Update Secret Value on Secret Object
+			testenvInstance.Log.Info("Data in secret object", "data", secretStruct.Data)
+			modifiedHecToken := testenv.GetRandomeHECToken()
+			modifedValue := testenv.RandomDNSName(10)
+			updatedSecretData := map[string][]byte{
+				"hec_token":    []byte(modifiedHecToken),
+				"password":     []byte(modifedValue),
+				"pass4SymmKey": []byte(modifedValue),
+				"idxc_secret":  []byte(modifedValue),
+				"shc_secret":   []byte(modifedValue),
+			}
```

Comment:
NIT: can be created into a function. Since its called in each test case. To modify the secret object and return ```updatedSecretData```

Reply:
Yeah that's a good idea I will make the change.